lib/core/test/unit/Cardano/Pool/MetricsSpec.hs

```diff
@@ -142,11 +176,24 @@ instance Arbitrary Block where
      arbitrary = genericArbitrary
      shrink = genericShrink
 
-instance Arbitrary (Quantity "block" Word32) where
-     arbitrary = Quantity . fromIntegral <$> (arbitrary @Word32)
-
-instance Arbitrary (Quantity any Word64) where
-     arbitrary = Quantity . fromIntegral <$> (arbitrary @Word64)
+deriving via Word32 instance (Arbitrary (Quantity "block" Word32))
+deriving via (Small (Word64)) instance (Arbitrary (Quantity "block" Word64))
+deriving via Lovelace instance (Arbitrary (Quantity "lovelace" Word64))
+
+-- TODO: Move to a shared location for Arbitrary newtypes
+newtype Lovelace = Lovelace Word64
+instance Arbitrary Lovelace where
+    shrink (Lovelace x) = map Lovelace $ shrink x
+    arbitrary = do
+        n <- choose (0, 100)
+        Lovelace <$> frequency
+            [ (50, return n)
+            , (25, return $ minLovelace - n)
```

Comment:
What is this supposed to mean :thinking: ? I believe that for `Word64`, this will simply loop back to `maxBound`. What's the intent behind this?

Reply:
Hm, oops, that should have been `maxBound - n`.

But I will remove this as I needed it to avoid overflow the test that is now not here.